tests/integration_tests/saas/test_adobe_campaign_task.py

```diff
@@ -0,0 +1,308 @@
+            "blackList",
+            "blackListAllLastModified",
+            "blackListEmail",
+            "blackListEmailLastModified",
+            "blackListFax",
+            "blackListFaxLastModified",
+            "blackListLastModified",
+            "blackListMobile",
+            "blackListMobileLastModified",
+            "blackListPhone",
+            "blackListPhoneLastModified",
+            "blackListPostalMail",
+            "blackListPostalMailLastModified",
+            "blackListPushnotification",
+            "ccpaOptOut",
+            "ccpaOptOutLastModified",
+            "created",
+            "cryptedId",
+            "domain",
+            "email",
+            "emailFormat",
+            "externalId",
+            "fax",
+            "firstName",
+            "gender",
+            "href",
+            "isExternal",
+            "lastModified",
+            "lastName",
+            "location",
+            "middleName",
+            "mobilePhone",
+            "phone",
+            "postalAddress",
+            "preferredLanguage",
+            "salutation",
+            "subscriptions",
+            "thumbnail",
+            "timeZone",
+            "title",
+        ],
+    )
+    assert_rows_match(
+        v[f"{dataset_name}:marketing_history"],
+        min_size=1,
+        keys=[
+            "PKey",
+            "age",
+            "birthDate",
+            "blackList",
+            "blackListAllLastModified",
+            "blackListEmail",
+            "blackListEmailLastModified",
+            "blackListFax",
+            "blackListFaxLastModified",
+            "blackListLastModified",
+            "blackListMobile",
+            "blackListMobileLastModified",
+            "blackListPhone",
+            "blackListPhoneLastModified",
+            "blackListPostalMail",
+            "blackListPostalMailLastModified",
+            "ccpaOptOut",
+            "ccpaOptOutLastModified",
+            "countBroadLogEvents",
+            "countSubHistoEvents",
+            "countryIsoA2",
+            "created",
+            "email",
+            "events",
+            "externalId",
+            "fax",
+            "firstName",
+            "gender",
+            "href",
+            "isExternal",
+            "kpisAndChart",
+            "lastModified",
+            "lastName",
+            "location",
+            "middleName",
+            "minBroadLogEvents",
+            "minSubHistoEvents",
+            "mobilePhone",
+            "phone",
+            "preferredLanguage",
+            "salutation",
+            "thumbnail",
+            "timeZone",
+            "title",
+        ],
+    )
+
+    x = graph_task.run_erasure(
+        privacy_request,
+        policy,
+        graph,
+        [adobe_campaign_connection_config],
+        {"email": erasure_email},
+        get_cached_data_for_erasures(privacy_request.id),
+    )
+
+    # Assert erasure request made to adobe_campaign_user
+    assert x == {
```

Comment:
i assume here the erasure request isn't completed within a timeframe that's reasonable to wait for in the test. if that's the case, then nothing to change; if it's not, then i'd say we should poll the API to make sure the expected data is actually deleted.

Reply:
That is correct, this creates an Adobe equivalent of a privacy request which then requires manual approval. In this test, we just verify that the privacy request is submitted to Adobe.